include/nbl/video/IGPUFence.h

```diff
@@ -31,6 +33,94 @@ class IGPUFence : public core::IReferenceCounted, public IBackendObject
     virtual ~IGPUFence() = default;
 };
 
+class GPUEventWrapper : public core::Uncopyable
```

Comment:
this needs to go to a different file and be done in 3 distinct flavours:
- fence
- semaphore
- polymorphic either fence or semaphore

Reply:
forget what I said, only a timeline semaphore (which we dont support) is waitable on the host

and waiting on an event doesnt mean shit, because you cannot delete/modify any vulkan object if its in a commandbuffer that is pending/executing, also it gives 0 guarantees about memory dependencies of any kind.